### PR TITLE
PLAT-204: implement unordered terminal method calls

### PR DESCRIPTION
**Pull request by bigbes**

### CODE REVIEW

File: ledger-core/v2/logicrunner/preprocessor/templates/wrapper.go.tpl [hunk 1]
```diff
@@ -263,7 +263,10 @@ func Initialize() insolar.ContractWrapper {
 		GetPrototype: INSMETHOD_GetPrototype,
 		Methods: insolar.ContractMethods{
 			{{ range $method := .Methods -}}
-					"{{ $method.Name }}": INSMETHOD_{{ $method.Name }},
+					"{{ $method.Name }}": insolar.ContractMethod{
+					    Func: INSMETHOD_{{ $method.Name }},
+					    Unordered: {{ $method.Immutable }},
```
Comment (aalekseev-ru) [inline ledger-core/v2/logicrunner/preprocessor/templates/wrapper.go.tpl:268]: It seems kinda strange. Do we now have 2 different types of call - unordered AND immutable? As I know, now we replace what we used to call 'immutable' with 'Unordered'. So I think we need to remove 'Immutable' from code base. And it is strange that we change the way wrapper is generated.

File: ledger-core/v2/runner/executor/builtin/builtin.go [hunk 2]
```diff
@@ -86,12 +87,26 @@ func (b *Runner) CallMethod(
 	}
 	contract := b.CodeRegistry[contractName]
 
-	methodFunc, ok := contract.Methods[method]
+	methodFuncObject, ok := contract.Methods[method]
```
Comment (aalekseev-ru) [inline ledger-core/v2/runner/executor/builtin/builtin.go:90]: If we rename it to methodObject we won't use something like return methodFunc.Func(data, args) (see above)

File: ledger-core/v2/logicrunner/builtin/builtin.go [hunk 3]
```diff
@@ -121,5 +121,5 @@ func (b *BuiltIn) CallMethod(ctx context.Context, callCtx *insolar.LogicCallCont
 		return nil, nil, errors.New("failed to find contracts method")
 	}
 
-	return methodFunc(data, args)
+	return methodFunc.Func(data, args)
```
Comment (aalekseev-ru) [inline ledger-core/v2/logicrunner/builtin/builtin.go:124]: maybe rename to methodObject?

File: ledger-core/v2/virtual/integration/method_test.go [hunk 4]
```diff
@@ -159,6 +162,104 @@ func TestVirtual_Method_WithoutExecutor(t *testing.T) {
 	}
 }
 
+func TestVirtual_Method_WithoutExecutor_Immutable(t *testing.T) {
```
Comment (aalekseev-ru) [inline ledger-core/v2/virtual/integration/method_test.go:165]: why it is still 'immutable'?

File: ledger-core/v2/logicrunner/preprocessor/templates/wrapper.go.tpl (hunk 1 above)
Comment (bigbes) [reply ledger-core/v2/logicrunner/preprocessor/templates/wrapper.go.tpl:268]: If it's need to be done - it should be different PR

File: ledger-core/v2/runner/executor/builtin/builtin.go (hunk 2 above)
Comment (bigbes) [reply ledger-core/v2/runner/executor/builtin/builtin.go:90]: sure

File: ledger-core/v2/logicrunner/builtin/builtin.go (hunk 3 above)
Comment (bigbes) [reply ledger-core/v2/logicrunner/builtin/builtin.go:124]: Old codebase, just supported compilation

File: ledger-core/v2/virtual/integration/method_test.go (hunk 4 above)
Comment (bigbes) [reply ledger-core/v2/virtual/integration/method_test.go:165]: sure

File: ledger-core/v2/logicrunner/builtin/builtin.go (hunk 3 above)
Comment (ruz) [reply ledger-core/v2/logicrunner/builtin/builtin.go:124]: https://insolar.atlassian.net/browse/PLAT-298

File: ledger-core/v2/logicrunner/preprocessor/templates/wrapper.go.tpl (hunk 1 above)
Comment (ruz) [reply ledger-core/v2/logicrunner/preprocessor/templates/wrapper.go.tpl:268]: https://insolar.atlassian.net/browse/PLAT-296

**Pull request merged**